data-prepper-plugins/kafka-source/README.md

```diff
@@ -0,0 +1,150 @@
+# Kafka source
+
+This is the Data Prepper Kafka source plugin that reads records from Kafka topic. It uses the consumer API provided by Kafka to read messages from the broker.
+
+The Kafka source plugin supports OpenSearch 2.0.0 and greater.
+
+## Usages
+
+The Kafka source should be configured as part of Data Prepper pipeline yaml file.
+
+## Configuration Options
+
+```
+log-pipeline:
+  source:
+    kafka:
+      bootstrap_servers:
+        - 127.0.0.1:9092,localhost:9093, 127.0.0.1:9094
+      topics:
```

Comment:
Incorporate latest YAML review comments as discussed for topics and others

Reply:
Except the topic related changes, all other changes have been resolved.

We removed the config value 'name' under the topics and assigned the topic-name directly to the -topic variable as shown below.
     ```
      topics:
        - topic: my-topic-2
          auth_type: SASL_OAUTH
          consumer:
              group_name: kafka-consumer-group-2
              group_id: DPKafkaProj-2
```
then we get the below exception:
org.opensearch.dataprepper.parser.ParseException: Failed to parse the configuration file D:\Projects\data-prepper-new\pipelines